parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMemoryManager.java

```diff
@@ -44,89 +44,146 @@
       "required int32 line;\n" +
       "required binary content;\n" +
       "}";
-  long expectPoolSize;
-  int rowGroupSize;
+  long expectedPoolSize;
   ParquetOutputFormat parquetOutputFormat;
-  CompressionCodecName codec;
   int counter = 0;
-  boolean firstRegister = true;
 
   @Before
-  public void setUp() {
-    GroupWriteSupport.setSchema(MessageTypeParser.parseMessageType(writeSchema),conf);
-    expectPoolSize = Math.round((double) ManagementFactory.getMemoryMXBean().getHeapMemoryUsage().getMax
-        () * MemoryManager.DEFAULT_MEMORY_POOL_RATIO);
-    rowGroupSize = (int) Math.floor(expectPoolSize / 2);
-    conf.setInt(ParquetOutputFormat.BLOCK_SIZE, rowGroupSize);
-    codec = CompressionCodecName.UNCOMPRESSED;
+  public void setUp() throws Exception {
+    parquetOutputFormat = new ParquetOutputFormat(new GroupWriteSupport());
+
+    GroupWriteSupport.setSchema(MessageTypeParser.parseMessageType(writeSchema), conf);
+    expectedPoolSize = Math.round((double)
+        ManagementFactory.getMemoryMXBean().getHeapMemoryUsage().getMax() *
+        MemoryManager.DEFAULT_MEMORY_POOL_RATIO);
+
+    long rowGroupSize = expectedPoolSize / 2;
+    conf.setLong(ParquetOutputFormat.BLOCK_SIZE, rowGroupSize);
+
+    // the memory manager is not initialized until a writer is created
+    createWriter(1).close(null);
   }
 
-  @After
-  public void tearDown() throws Exception{
-    FileUtils.deleteDirectory(new File("target/test"));
+  @Test
+  public void testMemoryManagerUpperLimit() {
+    // Verify the memory pool size
+    // this value tends to change a little between setup and tests, so this
+    // validates that it is within 5% of the expected value
+    long poolSize = ParquetOutputFormat.getMemoryManager().getTotalMemoryPool();
+    Assert.assertTrue("Pool size should be within 5% of the expected value",
+        Math.abs(expectedPoolSize - poolSize) < (long) (expectedPoolSize * 0.05));
   }
 
   @Test
   public void testMemoryManager() throws Exception {
-    //Verify the adjusted rowGroupSize of writers
+    long poolSize = ParquetOutputFormat.getMemoryManager().getTotalMemoryPool();
+    long rowGroupSize = poolSize / 2;
+    conf.setLong(ParquetOutputFormat.BLOCK_SIZE, rowGroupSize);
+
+    Assert.assertTrue("Pool should hold 2 full row groups",
+        (2 * rowGroupSize) <= poolSize);
+    Assert.assertTrue("Pool should not hold 3 full row groups",
+        poolSize < (3 * rowGroupSize));
+
```

Comment:
These tests validate assumptions about the current settings and environment. The idea is to catch environment problems early rather than complaining about a code failure.